doc/api/fs.md

```diff
@@ -5089,6 +5153,43 @@ characters directly to the `prefix` string. For instance, given a directory
 `prefix` must end with a trailing platform-specific path separator
 (`require('path').sep`).
 
+### `fsPromises.mkstemp(prefix[, options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `prefix` {string}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {Promise}
+
+Creates and opens a unique empty temporary file, and fulfills the `Promise` with
+an object that contains the created file path and a `FileHandle` object. A
+unique file name is generated by appending six random characters to the end of
+the provided `prefix`. Due to platform inconsistencies, avoid trailing `X`
+characters in `prefix`. Some platforms, notably the BSDs, can return more than
+six random characters, and replace trailing `X` characters in `prefix` with
+random characters.
+
+The optional `options` argument can be a string specifying an encoding, or an
+object with an `encoding` property specifying the character encoding to use.
+
+```js
+async function createRandomFile() {
+  const result = await fsPromises.mkstemp(path.join(os.tmpdir(), 'foo-'));
+  console.log(result.path);
+  // Prints: /tmp/foo-itXde2 or C:\Users\...\AppData\Local\Temp\foo-itXde2
+  await result.handle.close();
```

Comment:
```suggestion
  await result.handle.close();
  // Closes the file descriptor but does not remove the file.
```

@benjamingr ?

Reply:
@targos Question: do we want to `unlink` instead of the comment in the examples so that it's clear users should _generally_ clean up after temp files they create?